samples/java-protobuf-tracing/src/main/java/com/example/ControllerAction.java

```diff
@@ -0,0 +1,137 @@
+package com.example;
+
+import com.google.protobuf.Empty;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import kalix.javasdk.action.ActionCreationContext;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+public class ControllerAction extends AbstractControllerAction {
+
+  String url = "https://jsonplaceholder.typicode.com/posts/1";
+  HttpClient httpClient = HttpClient.newHttpClient();
+  public ControllerAction(ActionCreationContext creationContext) {}
+
+
+  @Override
+  public Effect<ControllerActionApi.MessageResponse> callSyncEndpoint(Empty empty) {
+    //Taking the already configured tracer. Such it will know where to export the spans
+    Optional<Tracer> tracerOpt = actionContext().getOpenTelemetryTracer();
+    String result;
+
+    if (tracerOpt.isPresent()){
+      result = callSyncService(tracerOpt.get());
+    } else {
+      try {
+        HttpResponse<String> response = callSyncService();
+        result = response.body();
+      } catch (IOException | InterruptedException e){
+        result = e.getMessage();
+      }
+    }
+    return effects().reply(ControllerActionApi.MessageResponse.newBuilder().setMessage(result).build());
+  }
+
+  private  HttpResponse<String> callSyncService() throws IOException, InterruptedException {
+    HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(url))
+            .build();
+    return httpClient.send(request, HttpResponse.BodyHandlers.ofString());
```

Comment:
Maybe better not to promote sync calls. We could keep only the async one. 

The reason is that Kalix users are building distributed microservices and async must be central concept all the time.

Reply:
Would be different if we were already on Java 21 and supporting Loom.

Reply:
Okay. This makes sense. I thought about this but failed to formulate the question. 
I'll delete the sync.

Reply:
done